src/Symfony/Component/PasswordHasher/Tests/Hasher/UserPasswordHasherTest.php

```diff
@@ -93,3 +127,10 @@ public function testNeedsRehash()
         $this->assertFalse($passwordHasher->needsRehash($user));
     }
 }
+
+abstract class TestPasswordAuthenticatedUser implements UserInterface, LegacyPasswordAuthenticatedUserInterface
+{
+    abstract public function getPassword(): ?string;
+
+    abstract public function getSalt(): ?string;
```

Comment:
```suggestion
```

Those are already declared on the interfaces you implement. There should be no need to redeclare them.

Reply:
Both UserInterface and the new PasswordAuthenticatedUserInterface expose the same methods, with a different signature (the former has no return type). This is needed to be able to mock it (signature mismatch otherwise)

Reply:
I see, thanks!

Reply:
Fixed by inverting the two `implements` (making the new interface first) :)